### ISSUE TITLE
Single quotes not escaped properly

### ISSUE THREAD

**Issue by jasonhansel** [kylekatarnls/jade-php#4]
Putting a single quote inside the value of an attribute causes problems, because htmlspecialchars is called without the `ENT_QUOTES` flag; see [the PHP docs](http://php.net/manual/en/function.htmlspecialchars.php) for further details.

**Comment by kylekatarnls** [comment]
What kind of problems? Please put an example of what you get, what you want to get and why. And please see if the javascript jade engine has a different behaviour.

**Comment by MicMonen** [comment]
+mixin('Name', 'Value')(attribute='Hello World') returns null
+mixin('Name', 'Value')(attribute="Hello World") returns "Hello World" (double quoted)
+mixin('Name', 'Value')(attribute=Hello World) returns Hello

You want to have just Hello World returned (singel quoted)

**Comment by jasonhansel** [comment]
Sorry I just got around to making a proper test case of the issue I've experienced.

Suppose you have the following code:

```
$data['title'] = "This is a 'test'";
$jade = new Jade();

$template = <<<JADE
a(href=title)
JADE;

echo $jade->render($template, $data);
```

You would probably expect this to produce an `<a>` tag whose `href` attribute is `This is a 'test'`. Instead, this just sets it to `This is a`, since the `'`s in the attribute are not escaped.

In other words, the expected output is `<a href='This is a &quot;test&quot;"></a>` but the real output is `<a href='This is a 'test''></a>`.

This presents a serious security concern -- it opens up an XSS vulnerability in any app using this library, since the user could (for instance) use this bug to set an element's `onclick` or `onmouseover` attributes to something malicious.

**Comment by kylekatarnls** [comment]
OK, also reproduced in pure Jade:
```a(title="test 'test'") Foo```

I will fix it soon.

**Comment by kylekatarnls** [comment]
Hi, just to say, with this issue I found another one in the original library. It seems that any escape is done to any attribute (attr="value" is the same as attr!="value"). So there is no htmlspecialchars at all on attributes currently neither on this lib nor on other forks. I will add the escape. Right now please ensure you do not pass use input on any attributes or add manually htmlspecialchars($attrValue, ENT_QUOTES).

**Comment by phibao37** [comment]
Could I ask a related question
Any option to change from single quote to double quote around attributes?
I see that all compiled attributes are single-quoted:
`html(lang="en")` => `<html lang='en'>`
`.row` => `<div class='row'>`
Maybe all web browsers understand single quote, but I still prefer double quote

**Comment by kylekatarnls** [comment]
Should be fixed now (use 1.5.1 version or newer). And the single/double quote option is now available.
